core/lib/via_btc_client/examples/data_inscription_example.rs

```diff
@@ -0,0 +1,682 @@
+                    .unwrap(),
+            };
+
+            utxos.push((out_point, tx_out));
+            println!("found utxo: {:?}", txid);
+        }
+    }
+
+    utxos
+}
+
+fn get_user_input<C: Signing>(
+    secp: &Secp256k1<C>,
+) -> (SecretKey, WPubkeyHash, Address, Keypair, String, bool) {
+    let mut render_config = RenderConfig::default();
+    render_config.prompt_prefix = Styled::new(">").with_fg(Color::LightGreen);
+    render_config.prompt = StyleSheet::new().with_fg(Color::LightMagenta);
+
+    let user_wif_prv = Text::new("Enter your private key (WIF): ")
+        .with_render_config(render_config)
+        .prompt()
+        .unwrap();
+
+    let user_wif_prv = user_wif_prv.trim();
+
+    let private_key = PrivateKey::from_wif(user_wif_prv).expect("Invalid Private Key WIF format");
+    let sk = private_key.inner;
+
+    let pk = bitcoin::PublicKey::new(sk.public_key(secp));
+    let wpkh = pk.wpubkey_hash().expect("key is compressed");
+
+    let compressed_pk = CompressedPublicKey::from_private_key(secp, &private_key).unwrap();
+    let address = Address::p2wpkh(&compressed_pk, Network::Testnet);
+
+    let keypair = Keypair::from_secret_key(secp, &sk);
+
+    println!("Your address: {}", address);
+
+    let multiline_content = r#"
+    Please check the printed address above and make sure it is correct.
+    if it's not press ctrl+c to exit and try again.
+
+    Enter the data you want to inscribe (string or hexstring): 
+    "#;
+    let data = Text::new(multiline_content)
+        .with_render_config(render_config)
+        .prompt()
+        .unwrap();
+
+    let trimmed_data = data.trim().to_string();
+
+    let is_proof = trimmed_data.as_bytes().len() > 520;
+
+    (sk, wpkh, address, keypair, trimmed_data, is_proof)
+}
+
+fn greeting() {
+    let content = r#"
+    
+    Welcome! 
+    
+    This is an CLI application that walks you through 
+    inscribing arbitrary data into the Bitcoin testnet.
+
+    **Please before continuing make sure you have done the following:**
+    
+    1- Install electrum wallet (https://electrum.org/#download)
+    And run it in testnet mode with using the following command:
+    Linux: electrum --testnet
+    Mac: /Applications/Electrum.app/Contents/MacOS/run_electrum --testnet
+
+    2- create a p2wpkh wallet (this is the default wallet type in electrum).
+    
+    3- get some testnet coins.
+    
+    Faucet Links:
+        https://bitcoinfaucet.uo1.net/
+        https://coinfaucet.eu/en/btc-testnet/
+    
+    when you are ready, press enter to continue...
+    "#;
+
+    let mut render_config = RenderConfig::default();
+    render_config.prompt_prefix =
+        Styled::new("***********************************************************")
+            .with_fg(Color::LightRed);
+    render_config.prompt = StyleSheet::new().with_fg(Color::Grey);
+
+    let res = Text::new(content)
+        .with_render_config(render_config)
+        .prompt();
+
+    match res {
```

Comment:
in this cases (when you need to handle only one option)
you can use
```
res.is_err() {
...
}
```
but if we use anyhow error handling here also, then we can just get rid of it at all.
```
    let res = Text::new(content)
        .with_render_config(render_config)
        .prompt()?;
```